### ISSUE TITLE
Literal type not accepted as key when constructing TypedDict

### ISSUE THREAD

**Issue by JukkaL** [python/mypy#7644]
@b0g3r reported an issue with TypedDict keys not being accepted with literal types. Example code from https://github.com/python/mypy/issues/6262#issuecomment-538520768:

```py
from mypy_extensions import TypedDict
from typing_extensions import Literal, Final


class MyDict(TypedDict):
    field_name: int

d: MyDict
d = {'field_name': 1}  # OK

name_in_var = 'field_name'
d = {name_in_var: 1}  # Correct error

name_in_literal2: Literal['field_name'] = 'field_name'
d = {name_in_literal2: 1}  # Unexpected error

name_in_final: Final = 'field_name'
d = {name_in_final: 1}  # Unexpected error
```

**Comment by b0g3r** [comment]
Oh no, I wanted to implement it :(

But thanks anyway!
